modules/internal/ChapelBase.chpl

```diff
@@ -307,11 +307,31 @@ module ChapelBase {
   //
   // % on primitive types
   //
-  inline proc %(a: int(?w), b: int(w)) return __primitive("%", a, b);
-  inline proc %(a: uint(?w), b: uint(w)) return __primitive("%", a, b);
+  inline proc %(a: int(?w), b: int(w)) {
+    if (chpl_checkDivByZero) then
+      if b == 0 then
+        halt("Attempt to divide by zero");
```

Comment:
I originally thought this would be simpler, but now I've changed my mind. I think clarifying the operation would be an improvement. For reference, Python does this:

```
ZeroDivisionError: integer division or modulo by zero
```

and clang does this:

```
warning: remainder by zero is undefined [-Wdivision-by-zero]
```